examples/federated_statistics/image_stats/image_stats_job/custom/image_histogram.py

```diff
@@ -0,0 +1,121 @@
+
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.abstract.statistics_spec import Statistics, Histogram, Feature, DataType, Bin, HistogramType
+
+
+class ImageHistogram(Statistics):
+
+    def __init__(
+            self,
+            data_root: str = "/tmp/nvflare/data",
+            data_list_key: str = "data"
+    ):
+        """local image histogram generator for federated analysis.
+
+        Args:
+            data_root: directory with local image data.
+            data_list_key: data list key to use.
+        Returns:
+            a Shareable with the computed local histograms`
+        """
+        super().__init__()
+        self.data_list_key = data_list_key
+        self.data_root = data_root
+        self.data_list = None
+        self.client_name = None
+
+        self.loader = None
+        self.failure_images = 0
+
+
+    def initialize(self, parts: dict, fl_ctx: FLContext):
+        self.client_name = fl_ctx.get_identity_name()
+        self.loader = LoadImage()
+        self.loader.register(ITKReader())
+        self._load_data_list(self.client_name, fl_ctx)
+
+        if self.data_list is None:
+            raise ValueError("data is not loaded. make sure the data is loaded")
+
+    def _load_data_list(self, client_name, fl_ctx: FLContext) -> bool:
+        dataset_json = glob.glob(os.path.join(self.data_root, client_name + "*.json"))
+        if len(dataset_json) != 1:
+            self.log_error(
+                fl_ctx, f"No unique matching dataset list found in {self.data_root} for client {client_name}"
+            )
+            return False
+        dataset_json = dataset_json[0]
+        self.log_info(fl_ctx, f"Reading data from {dataset_json}")
+        data_list = load_decathlon_datalist(
+            data_list_file_path=dataset_json, data_list_key=self.data_list_key, base_dir=self.data_root
+        )
+        self.data_list = {"train": data_list}
+
+        self.log_info(fl_ctx, f"Client {client_name} has {len(self.data_list)} images")
+        return True
+
+    def features(self) -> Dict[str, List[Feature]]:
+        return {"train": [Feature("density", DataType.FLOAT)]}
+
+    def count(self,
+              dataset_name: str,
+              feature_name: str) -> int:
+        image_paths = self.data_list[dataset_name]
+        return len(image_paths)
+
+
+    def failure_count(self,
+              dataset_name: str,
+              feature_name: str) -> int:
+
+        return self.failure_images
+
+    def histogram(self, dataset_name: str, feature_name: str, num_of_bins: int, global_min_value: float,
+                  global_max_value: float) -> Histogram:
+        histogram_bins: List[Bin] = []
+        histogram = np.zeros((num_of_bins,), dtype=np.int64)
+
+        for i, entry in enumerate(self.data_list[dataset_name]):  # TODO: use multi-processing
+            file = entry.get("image")
+            try:
+                img, meta = self.loader(file)
```

Comment:
this `img, meta = self.loader(file)` has nothing to do with calculation.
suggest move these loading logic into `initialize`

Reply:
can't do this in initialize because of memory issue. got it